circe-compat/test/src/nrktkt/ninny/compat/CirceCompatSpec.scala

```diff
@@ -0,0 +1,66 @@
+package nrktkt.ninny.compat
+
+import nrktkt.ninny.ast._
+import nrktkt.ninny._
+import io.circe._
+import io.circe.generic.semiauto._
+import io.circe.syntax._
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should
+import org.scalatest.OptionValues
+import org.scalatest.TryValues
+import CirceCompat._
+
+import scala.util.Failure
+
+
+class CirceCompatSpec
+    extends AnyFlatSpec
+    with should.Matchers
+    with OptionValues
+    with TryValues {
+
+  case class Example1(foo: String, bar: Seq[Int])
+  object Example1 {
+    implicit val encoder: Encoder[Example1] = deriveEncoder[Example1]
+    implicit val decoder: Decoder[Example1] = deriveDecoder[Example1]
+  }
+
+  case class Example2(foo: String, bar: Seq[Int])
+  object Example2 {
+    implicit val toJson   = ToJson.auto[Example2]
+    implicit val fromJson = FromJson.auto[Example2]
+  }
+
+  val ex1 = Example1("baz", Seq(1, 2, 3))
+  val ex2 = Example2("baz", Seq(1, 2, 3))
+
+  val ex1json = obj("foo" ~> "baz", "bar" ~> Seq(1, 2, 3))
+  val ex2json = io.circe.Json.obj(
+    "foo" -> io.circe.Json.fromString("baz"),
+    "bar" -> io.circe.Json.fromValues(Seq(1,2,3).map(io.circe.Json.fromInt))
+  )
+
+  "Circe typeclasses" should "write ninny json" in {
+    val json = ex1.toSomeJson
+    json shouldEqual ex1json
+  }
+
+  it should "read ninny json" in {
+    val objekt = ex1json.to[Example1]
+    objekt match {
+      case Failure(e: DecodingFailure) => println(e.printStackTrace())
+    }
+    objekt shouldEqual ex1
+  }
```

Comment:
This is not passing. I'm still trying to figure out. @nrktkt

Reply:
`objekt` is a `Try` and `ex1` is an `Example1`. so the comparison fails
```suggestion
    val objekt = ex1json.to[Example1]
    objekt match {
      case Failure(e: DecodingFailure) => println(e.printStackTrace())
    }
    objekt.success.value shouldEqual ex1
  }
```

Reply:
the Try is a Failure, i was trying to print the stacktrace.

Reply:
It turns out that `NinnyToCirce` and `CirceToNinny` put toegher, is breaking Circe itself's decoding functionality, 

i.e. a Circe json object's `Decoder` is implicitly converted to Ninny's `FromJson`, then implicitly converted to `Decoder`, over and over.

So I think it would be better to separate `CirceToNinny` and `NinnyToCirce`

E.g.

```
import CirceCompat._

case class Person(name: String, age: Int)
object Person {
  implicit val encoder = deriveEncoder[Person]
  implicit val decoder = deriveDecoder[Person]
}

val json = Person("alice", 30).asJson

// Decoder[Person] is implicitly converted to FromJson[Person] by CirceCompat
// then FromJson is converted to Decoder, so on so forth
json.to[Person]

```